storage/pin/pin_test.go

```diff
@@ -0,0 +1,509 @@
+// Copyright 2019 The Swarm Authors
+// This file is part of the Swarm library.
+//
+// The Swarm library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The Swarm library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the Swarm library. If not, see <http://www.gnu.org/licenses/>.
+
+package pin
+
+import (
+	"bytes"
+	"context"
+	"encoding/hex"
+	"io/ioutil"
+	"mime"
+	"os"
+	"path/filepath"
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/ethersphere/swarm/api"
+	"github.com/ethersphere/swarm/chunk"
+	"github.com/ethersphere/swarm/state"
+	"github.com/ethersphere/swarm/storage"
+	"github.com/ethersphere/swarm/storage/feed"
+	"github.com/ethersphere/swarm/storage/localstore"
+	"github.com/ethersphere/swarm/testutil"
+)
+
+// TestPinRawUpload pins a RAW file and unpin it multiple times
+func TestPinRawUpload(t *testing.T) {
+	p, f, closeFunc := getPinApiAndFileStore(t)
+	defer closeFunc()
+
+	data := testutil.RandomBytes(1, 10000)
+	hash := uploadFile(t, f, data, false)
+
+	// test pin and unpin
+	pinUnpinAndFailIfError(t, p, hex.EncodeToString(hash), 3, true)
+}
+
+// TestPinRawUploadEncrypted pins a encrypted RAW file and unpin it multiple times
```

Comment:
if you pin a file and then unpin it multiple files, shouldnt that result in an error? the test says "failIfError" but i'm guessing you're not expecting an error.... seems a bit counterintuitive for me

Reply:
Nope.. it tries to pin the file 3 times and unpin it 3 times.. why should it give error? 
The test will fail if there is any error during pinning / unpinning or during subsequent checks of the chunks and index.

Reply:
cool thx